app/src/main/java/com/android/sample/ui/activity/EditActivity.kt

```diff
@@ -518,47 +519,56 @@ fun EditActivityScreen(
                         startTime?.isNotEmpty() ?: false &&
                         duration.isNotEmpty(),
                 onClick = {
-                  if (!hourDateViewModel.isBeginGreaterThanEnd(
+                  // we don't want to allow creators to schedule start time of activities 1 hour
+                  // before they start
+                  if (dueDate.toDate().time - System.currentTimeMillis() <
+                      TimeUnit.HOURS.toMillis(1)) {
+                    Toast.makeText(
+                            context,
+                            "Activity must be scheduled at least 1 hour in advance",
+                            Toast.LENGTH_SHORT)
+                        .show()
+                  } else if (!hourDateViewModel.isBeginGreaterThanEnd(
                       startTime ?: "00:00", duration ?: "00:01")) {
                     Toast.makeText(
                             context, "Start time must be before end time", Toast.LENGTH_SHORT)
                         .show()
-                    return@Button
```

Comment:
Is removing this return expected ?

Reply:
It worked before but it didn't anymore due to dueDate not being updated correctly (probably someone else edited that). I took this opportunity to refactor what we have and use return@Button on each condition we have. Will do the same for edit activities